app/main/views/suppliers.py

```diff
@@ -119,6 +119,38 @@ def view_signed_agreement(supplier_id, framework_slug):
     )
 
 
+@main.route('/suppliers/agreements/<agreement_id>/on-hold', methods=['POST'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def put_signed_agreement_on_hold(agreement_id):
+    data_api_client.put_signed_agreement_on_hold(agreement_id, current_user.email_address)
+
+    organisation = request.args['organisation']
```

Comment:
Passing in the organisation as a parameter on the form target seems odd to me.  Two options that I think would be better would be:
1. Pass it in a hidden form field from the page (`input type="hidden"`)
2. Do two API fetches in the POST route here - first get the FrameworkAgreement and then the SupplierFramework

I think option 1 would be OK as the value is only used for the flash messaging, so we can probably do without the API fetches.